### PR TITLE
Fix rock for MLflow 1.25

### PR DESCRIPTION
**Pull request by misohu**
Closes: https://github.com/canonical/base-mlflow/issues/15 

We are moving to use rock only instead of the Docker image. I have set the secrets to point to correct docker user. 

**Note:** I am installing MLFLow in Rock directly rather then installing it from requirements as that was the way we used in the Dockerfile.

This rock has been also tested from my personal dockerhub account directly in mlflow operator in this [PR ](https://github.com/canonical/mlflow-operator/pull/255) (so we know its working).

### CODE REVIEW

**Comment by kimwnasptd** [comment]
Could we also update the `README.md` of this repo since we are bumping the version?

File: tests/__pycache__/test_rock.cpython-38-pytest-8.3.2.pyc
Comment (orfeas-k) [inline tests/__pycache__/test_rock.cpython-38-pytest-8.3.2.pyc:1]: We don't need to commit this. We could remove this and also add a [gitignore file](https://github.com/canonical/kubeflow-rocks/blob/main/.gitignore)

File: .github/workflows/on_pull_request.yaml
Comment (orfeas-k) [inline .github/workflows/on_pull_request.yaml:1]: Should we test with the latest supported (1.29)?

File: README.md [hunk 1]
```diff
@@ -1,39 +1,39 @@
-# Ubuntu 22.04 based MLflow image v2.1.1
```
Comment (kimwnasptd) [inline README.md:1]: Could we also add the badges for the CI, similarly to 
https://github.com/canonical/training-operator-rock/blob/main/README.md

File: README.md [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
+To build the ROCK image manually:
+```bash
 rockcraft pack
 ```
 
-To copy resulting image `base-mflow_v2.1.1_amd64.rock` to Docker:
+To copy the resulting image `base-mlflow_v2.15.0_amd64.rock` to Docker:
+```bash
+sudo skopeo --insecure-policy copy oci-archive:base-mlflow_v2.15.0_amd64.rock docker-daemon:base-mlflow_v2.15.0_amd64.rock:rock
 ```
```
Comment (orfeas-k) [inline README.md:32]: Should we replace those commands with the new tox environments? This way, we won't need to maintain those too.

File: README.md [hunk 3]
```diff
@@ -1,39 +1,39 @@
...
+# Ubuntu 22.04 based MLflow image v2.15.0
+This repository regularly builds a new [MLflow](https://github.com/mlflow/mlflow) image based on LTS Ubuntu 22.04. After the new image is built, the CI runs simple tests against the image and runs the [trivy](https://github.com/aquasecurity/trivy) vulnerability scan. 
 
-```
-FROM charmed/base-mlflow:latest
+The image is currently published [here](https://hub.docker.com/r/charmedkubeflow/mlflow)
+You can use this image seamlessly as any other Python Docker image by specifying it as a base with:
+
+```dockerfile
+FROM charmedkubeflow/mlflow:latest
```
Comment (orfeas-k) [inline README.md:9]: Is the image name also changing? Won't it be `base-mlflow` anymore?

File: README.md [hunk 4]
```diff
@@ -1,39 +1,39 @@
-# Ubuntu 22.04 based MLflow image v2.1.1
-This repository regularly builds new [Mlflow](https://github.com/mlflow/mlflow) image base on LTS Ubuntu 22.04. The process weekly downloads the upstream MLflow v2.1.1 and changes the base for Docker image with the Dockerfile you can see in this repository. After the new image is built the ci runs simple tests against the image and runs the [trivy](https://github.com/aquasecurity/trivy) vulnerability scan. The results are published to the security tab in the repository (only visible to admins). 
 
-The image is currently published [here](https://hub.docker.com/r/charmed/base-mlflow)
-You can use this image seamlesly as any other python Docker image by specifying it as a base with.
+# Ubuntu 22.04 based MLflow image v2.15.0
+This repository regularly builds a new [MLflow](https://github.com/mlflow/mlflow) image based on LTS Ubuntu 22.04. After the new image is built, the CI runs simple tests against the image and runs the [trivy](https://github.com/aquasecurity/trivy) vulnerability scan.
```
Comment (orfeas-k) [inline README.md:3]: I understand that the part about the CI is not entirely true. And also, it's not clear to me when what this happens. I mean if we refer to the PR CI, then this builds, scans, tests and also publishes the image.

**Review by orfeas-k — COMMENTED** [review]
Good job @misohu! I have the following questions:
* Do we use `mlflow.yaml`? If yes, right now it looks out of date. If not, we can remove it.
* In a daily ([13 feb 2024](https://docs.google.com/document/d/1WYWf8BEbB_nPasVwBWaDhhmnu461hZdFNVSAA4an1-s/edit#heading=h.n0uyh4gpbq41)), we had decided to use nested folders for single-rock repositories, thus I think we should refactor this one as well (directory name should be the same as the rock name). WDYT?

File: README.md (hunk 3 above)
Comment (misohu) [reply README.md:11]: sorry it always was mlflow ... I fixed it ... just the repo is called base-mlflow

**Review by orfeas-k — APPROVED** [review]
Good job @misohu :)

**Comment by kimwnasptd** [comment]
Do we still need the Dockerfile now that we converted to `rockcraft.yaml` file?

**Comment by misohu** [comment]
Hey @kimwnasptd @orfeas-k, I have removed the Docker image dependancies ... because of that I could move the mlflow to requirements. I have pinned requirements and added update-requirements tox. Now I am testing the image in the charm here https://github.com/canonical/mlflow-operator/actions/runs/10301193187/job/28512189629

**Review by orfeas-k — APPROVED** [review]
Good job once more!

**Review by kimwnasptd — APPROVED** [review]
Looks good Michal!

**Pull request merged**